crates/dbsp/src/trace/cursor/saturating_cursor.rs

```diff
@@ -0,0 +1,247 @@
+use std::marker::PhantomData;
+
+use crate::{
+    dynamic::{DataTrait, Erase, Factory},
+    trace::Cursor,
+    DynZWeight, Timestamp,
+};
+
+pub struct SaturatingCursor<'a, K, V, T, const SATURATE: bool>
+where
+    K: DataTrait + ?Sized,
+    V: DataTrait + ?Sized,
+    T: Timestamp,
+{
+    cursor: Box<dyn Cursor<K, V, T, DynZWeight> + 'a>,
+    on_phantom_key: bool,
+    phantom_key: Box<K>,
+    phantom_val: Box<V>,
```

Comment:
I would not call this phantom_val, I don't think it has the same meaning as the standard PhantomData

Reply:
I can change it to `ghost`.